lib/serializer.js

```diff
@@ -43,3 +44,24 @@ Serializer.prototype.stringToByteArray = function (msg/* : string */) {
 Serializer.prototype.byteArrayToString = function (bytes/* : Uint8Array */) {
   return this.api.GenericMessage.decode(bytes).message
 }
+
+/**
+ * Serializes client sync credentials for accessing AWS and S3.
+ * @param {{aws: {{accessKeyId: <string>, secretAccessKey: <string>, sessionToken: <string>, expiration: <string>}}}, s3Post: {{bucket: <string>, postData: {{accessKeyId: <string>, secretAccessKey: <string>, sessionToken: <string>, expiration: <string>}}}}}} credentials
+ * @returns {Uint8Array}
+ */
+Serializer.prototype.credentialsToByteArray = function (credentials) {
+  return this.api.Credentials.encode({
+    aws: this.api.Credentials.lookup('Aws').create(credentials.aws),
+    s3Post: this.api.Credentials.lookup('S3Post').create(credentials.s3Post)
+  }).finish()
+}
+
+/**
+ * Deserializes client sync credentials for accessing AWS and S3.
+ * @param {Uint8Array} bytes
+ * @returns {{aws: <Uint8Array>, s3Post: <Uint8Array>}}
```

Comment:
surprised that this deserializes to Uint8Array in the fields instead of the credentials object